wolfcrypt/src/sp_int.c

```diff
@@ -1033,19 +1056,20 @@ int sp_add_d(sp_int* a, sp_int_digit d, sp_int* r)
         }
         if (i == a->used) {
             r->used++;
-            r->dp[i] = 1;
+            if (i < SP_INT_DIGITS)
+                r->dp[i] = 1;
+            else
+                return MP_VAL;
         }
     }
     r->dp[0] = t;
-    if (r != a) {
-        for (++i; i < a->used; i++) {
-            r->dp[i] = a->dp[i];
-        }
-    }
+    for (++i; i < a->used; i++)
```

Comment:
Did you intend to remove the `if (r != a)` check? Still seems like a valid case.

Reply:
@dgarske, I think it is a redundant check. 
@SparkiDev, please review?

Reply:
The check is to avoid copying when not needed.
Check not redundant for a public API.